src/main/java/ch/njol/skript/util/BlockUtils.java

```diff
@@ -114,5 +115,29 @@ public static BlockData createBlockData(String dataString) {
 			return null;
 		}
 	}
+
+	/**
+	 * Get the string version of a block, including type and location.
+	 * ex: 'stone' at 1.5, 1.5, 1.5 in world 'world'
+	 *
+	 * @param block Block to get string of
+	 * @param flags
+	 * @return String version of block
+	 */
+	@Nullable
+	public static String blockToString(Block block, int flags) {
+		String type = ItemType.toString(block, flags);
+		Location location = getLocation(block);
+		if (location == null) {
+			return null;
+		}
+
+		double x = location.getX();
+		double y = location.getY();
+		double z = location.getZ();
+		String world = location.getWorld().getName();
+
+		return String.format("'%s' at %s, %s, %s in world '%s'", type, x, y, z, world);
```

Comment:
watch out for the performance of this method, String#format can in fact be quite expensive in some cases, I imagine users won't do it often enough that it'd become an issue though